### ISSUE TITLE
clean.cmd --all start a build instead of doing a clean

### ISSUE THREAD

**Issue by weshaggard** [dotnet/buildtools#1080]
While intended to do a `clean.cmd -all` I typed `clean.cmd --all` (I guess I've been typing too many git commands recently where that is the convention).  When I do that I get:

```
> .\clean.cmd --all
Stop VBCSCompiler.exe execution.
Installing dotnet cli...
Restoring BuildTools version 1.0.26-prerelease-00828-03...
Initializing BuildTools...
Updating CLI NuGet Frameworks map...
Done initializing tools.
Running: D:\git\corefx\Tools\msbuild.cmd /nologo /verbosity:minimal /clp:Summary /maxcpucount /nodeReuse:false /l:BinCla
shLogger,Tools\net45\Microsoft.DotNet.Build.Tasks.dll;LogFile=binclash.log  /flp:v=normal;LogFile=clean.log
  [21:11:47.91] Verifying all auto-upgradeable dependencies...
  [21:11:49.14] Verifying all auto-upgradeable dependencies... Done.
  [21:11:49.14] Restoring all packages...
```

instead of doing any clean it actually started to do a build. I suspect this is because everything beyond "--" is ignored by the parser but I'm not sure why the "all" part was completely thrown out instead of being passed to the msbuild command and thus causing it to fail.

cc @maririos

**Comment by MattGal** [comment]
There are a few problems here.  I've prepared a couple fixes but at some point, we'll also want to also get the usage text out of the clean.cmd file since this is a recipe for getting out of synch with what's actually in the config.json file, either by adding new aliases or deleting old ones.

1) PR : https://github.com/dotnet/corefx/pull/12344 
While it would be much preferred to get all the values from config.json and run through Run.exe, this one's needed because we clearly want to be able to scorch EVERYTHING, including Run.exe / the tools directory.  So, one dash, two dashes, fifteen dashes, any casing... if they say clean all it should clean all.

2) PR: https://github.com/dotnet/buildtools/pull/1087 
It's clear we need the ability to have default aliases for certain commands in the config.json file;  This case is a great example, because when the flags that are needed specify build targets and are all missing, we get a default target (build) that is undesirable.
It would be very easy to error out if there's 1) no default alias and 2) aliases exist but I did not do that here because I have not explored all code paths yet and there may be a case where a command defines aliases but is still useful without specifying any of them.
